dask/dataframe/core.py

```diff
@@ -3253,6 +3253,9 @@ def dot(self, other, meta=no_default):
         if not isinstance(other, _Frame):
             raise TypeError("The second operand must be a dask array or dask dataframe")
 
+        if isinstance(self, DataFrame):
+            raise NotImplementedError("DataFrame.dot is not implemented yet, sorry!")
```

Comment:
Does `DataFrame.dot` not work for any reasonable `other` input? For example, what if `other` is a Dask Series? 

Alternatively, I wonder how difficult it would be to make `DataFrame.dot` work with `DataFrame` inputs